modules/gapi/test/common/gapi_core_tests_inl.hpp

```diff
@@ -14,6 +14,327 @@
+        boundingBox.height += dy;
+    }
+}
+
+std::tuple<cv::Rect, float, float, float> unitedParseRef(const float* it,
+                                                         const cv::Size& in_size) {
+    float image_id   = it[0];
+    float label      = it[1];
+    float confidence = it[2];
+    float rc_left    = it[3];
+    float rc_top     = it[4];
+    float rc_right   = it[5];
+    float rc_bottom  = it[6];
+
+    cv::Rect rc;  // map relative coordinates to the original image scale
+    rc.x      = static_cast<int>(rc_left   * in_size.width);
+    rc.y      = static_cast<int>(rc_top    * in_size.height);
+    rc.width  = static_cast<int>(rc_right  * in_size.width)  - rc.x;
+    rc.height = static_cast<int>(rc_bottom * in_size.height) - rc.y;
+    return std::make_tuple(rc, image_id, confidence, label);
+}
+
+int randInRange(const int& start, const int& end)
+{
+    GAPI_Assert(start <= end);
+    return start + std::rand()%(end - start + 1);
+}
+
+cv::Rect generateBox(const cv::Size& in_sz)
+{
+    // Generated rectangle can reside outside of the initial image by $border pixels
+    constexpr int border = 10;
+    constexpr int minW = 16;
+    constexpr int minH = 16;
+    cv::Rect box;
+    box.width  = randInRange(minW, in_sz.width  + 2*border);
+    box.height = randInRange(minH, in_sz.height + 2*border);
+    box.x = randInRange(-border, in_sz.width  + border - box.width);
+    box.y = randInRange(-border, in_sz.height + border - box.height);
+    return box;
+}
+
+struct SSDitem
+{
+    float image_id;
+    float label;
+    float confidence;
+    float rc_left;
+    float rc_top;
+    float rc_right;
+    float rc_bottom;
+};
+
+SSDitem generateItem(const int& i, const cv::Size& in_sz)
+{
+    const auto normalize = [](int v, int range) { return static_cast<float>(v)/range; };
+
+    SSDitem it;
+    it.image_id = static_cast<float>(i);
+    it.label = static_cast<float>(randInRange(0, 9));
+    it.confidence = static_cast<float>(std::rand()) / RAND_MAX;
+    auto box = generateBox(in_sz);
+    it.rc_left   = normalize(box.x, in_sz.width);
+    it.rc_right  = normalize(box.x + box.width, in_sz.width);
+    it.rc_top    = normalize(box.y, in_sz.height);
+    it.rc_bottom = normalize(box.y + box.height, in_sz.height);
+
+    return it;
+}
+
+cv::Mat generateSSDoutput(const cv::Size& in_sz)
+{
+    constexpr int maxN = 200;
+    constexpr int objSize = 7;
+    std::vector<int> dims{1,1,maxN,objSize};
+    cv::Mat mat(dims, CV_32FC1);
+    auto data = mat.ptr<float>();
+
+    for (int i = 0; i < maxN; i++)
+    {
+        float* it = data + i*objSize;
+        auto ssdIt = generateItem(i, in_sz);
+        it[0] = ssdIt.image_id;
+        it[1] = ssdIt.label;
+        it[2] = ssdIt.confidence;
+        it[3] = ssdIt.rc_left;
+        it[4] = ssdIt.rc_top;
+        it[5] = ssdIt.rc_right;
+        it[6] = ssdIt.rc_bottom;
+    }
+
+    return mat;
+}
+
+cv::Mat generateYoloOutput(const int& num_classes)
```

Comment:
You may make the function void and have the output Mat as an argument by the reference

Reply:
Hmm, we should to use copyTo in case without return.

Reply:
Why? Can we just take `auto data = out_mat.ptr<float>();` as it is now in the code and write random numbers right in the output Mat?
But it may be insecure.. is it what you want to avoid?

Reply:
But it is not so important, and your way is easier and clearer